src/lib/fields_derivers_zkapps/fields_derivers_zkapps.ml

```diff
@@ -355,10 +355,15 @@ module Make (Schema : Graphql_intf.Schema) = struct
         ~of_string:sign_of_string
     in
     let ( !. ) = ( !. ) ~t_fields_annots:Currency.Signed_poly.t_fields_annots in
-    Currency.Signed_poly.Fields.make_creator obj ~magnitude:!.amount
-      ~sgn:!.sign_deriver
-    |> finish "BalanceChange"
-         ~t_toplevel_annots:Currency.Signed_poly.t_toplevel_annots
+    let balance_change obj =
+      Currency.Signed_poly.Fields.make_creator obj ~magnitude:!.amount
+        ~sgn:!.sign_deriver
+      |> finish "BalanceChange"
+           ~t_toplevel_annots:Currency.Signed_poly.t_toplevel_annots
+    in
+    needs_custom_js
```

Comment:
Could you add some information to PR description:

1. What was the format of balance change field before this PR
2. What will be after this PR
3. What's motivation of changing the format?

Reply:
done!